### PR TITLE
Raj av0.7.0 kernelupdates

### PR DESCRIPTION
**Pull request by rhornung67**
Added local accumulation variables to other kernels (based on work started by @artv3 and @MrBurmark) and reworked/cleaned up the code so it looks like it used to, more or less.

The next PR will be for a new release of the perf suite.

### CODE REVIEW

File: src/apps/ENERGY-OMPTarget.cpp [hunk 1]
```diff
@@ -161,35 +161,39 @@ void ENERGY::runOpenMPTargetVariant(VariantID vid)
...
-        ENERGY_BODY5;
-      });
-
-      RAJA::forall<RAJA::omp_target_parallel_for_exec<NUMTEAMS>>(
-        RAJA::RangeSegment(ibegin, iend), [=](int i) {
-        ENERGY_BODY6;
-     });
+      RAJA::region<RAJA::seq_region>( [=]() {
+
+        RAJA::forall<RAJA::omp_target_parallel_for_exec<NUMTEAMS>>(
```
Comment (rchen20) [inline src/apps/ENERGY-OMPTarget.cpp]: Not sure if this matters, but I merged https://github.com/LLNL/RAJA/pull/539 not long ago, changing "NUMTEAMS" to "ThreadsPerTeam".

Comment (rhornung67) [reply src/apps/ENERGY-OMPTarget.cpp]: Yes,  I'll update to a newer develop branch and change those for the release.

File: src/polybench/POLYBENCH_3MM-Cuda.cpp [hunk 2]
```diff
@@ -143,12 +143,13 @@ void POLYBENCH_3MM::runCudaVariant(VariantID vid)
     using EXEC_POL =
       RAJA::KernelPolicy<
         RAJA::statement::CudaKernelAsync<
-          RAJA::statement::For<0, RAJA::cuda_block_exec,
-            RAJA::statement::For<1, RAJA::cuda_thread_exec,
+          RAJA::statement::For<0, RAJA::cuda_block_x_loop,
+            RAJA::statement::For<1, RAJA::cuda_thread_y_loop,
```
Comment (MrBurmark) [inline src/polybench/POLYBENCH_3MM-Cuda.cpp]: Should this use cuda_block_y_loop, and cuda_thread_x_loop to more consistently use thread_x for adjacent threads.

File: src/polybench/POLYBENCH_ATAX-Cuda.cpp [hunk 3]
```diff
@@ -109,41 +112,74 @@ void POLYBENCH_ATAX::runCudaVariant(VariantID vid)
...
-    using EXEC_POL =
+    using EXEC_POL1 =
       RAJA::KernelPolicy<
         RAJA::statement::CudaKernelAsync<
-          RAJA::statement::For<0, RAJA::cuda_threadblock_exec<block_size>,
-            RAJA::statement::Lambda<0>,
-            RAJA::statement::For<1, RAJA::seq_exec,
-              RAJA::statement::Lambda<1>
+          RAJA::statement::Tile<0, RAJA::statement::tile_fixed<block_size>, RAJA::cuda_block_x_loop,
+            RAJA::statement::For<0, RAJA::cuda_thread_x_loop,
```
Comment (MrBurmark) [inline src/polybench/POLYBENCH_ATAX-Cuda.cpp]: We should consistently use cuda_thread_x_direct in fixed tiles.

File: src/polybench/POLYBENCH_3MM-Cuda.cpp (hunk 2 above)
Comment (rhornung67) [reply src/polybench/POLYBENCH_3MM-Cuda.cpp]: Yes, good catch! Thanks.

File: src/polybench/POLYBENCH_ATAX-Cuda.cpp (hunk 3 above)
Comment (rhornung67) [reply src/polybench/POLYBENCH_ATAX-Cuda.cpp]: Yes, thanks! I will commit changes shortly.

File: src/polybench/POLYBENCH_3MM-Cuda.cpp (hunk 2 above)
Comment (rhornung67) [reply src/polybench/POLYBENCH_3MM-Cuda.cpp]: Actually, I take that back. The RAJA variant matches the base CUDA variant.

Comment (rhornung67) [reply src/polybench/POLYBENCH_3MM-Cuda.cpp]: But, I will make the change you suggested here and in other kernels.  Thanks.

**Pull request closed without merging**